src/Service/PubliccodeService.php

```diff
@@ -0,0 +1,836 @@
+                return $logoFile['blob_id'];
+            }
+
+            // Return null if the logoFile response couldn't be decoded.
+            return null;
+        }//end if
+
+        // If the code comes here the logo is not found, so null can be returned.
+        return null;
+
+    }//end getLogoFileContent()
+
+
+    /**
+     * This function handles a github url to where the logo is placed in a repository. (https://github.com/OpenCatalogi/web-app/blob/development/pwa/src/assets/images/5-lagen-visualisatie.png)
+     * Option 3 of the handleLogo() function.
+     *
+     * @param array  $publiccodeArray The mapped publiccode array from the github api.
+     * @param Source $source          The github api source.
+     * @param array  $parsedLogo      The parsed logo that was given in the publiccode file. \Safe\parse_url($publiccodeArray['logo']);
+     * @param string $repositoryName  The fullname of the repository. /{owner}/{repository}
+     *
+     * @return string|null The updated logo with the download_url of the file contents with the path or null if not valid.
+     */
+    public function handleLogoFromGithub(array $publiccodeArray, Source $source, array $parsedLogo, string $repositoryName): ?string
+    {
+        // Explode the logo path with the repositoryName so the organization an repository name will be removed from the path.
+        $explodedPath = explode($repositoryName, $parsedLogo['path'])[1];
+
+        // The url of the logo from the github repository always has /blob/{branch} in the url.
+        // TODO: Check if this is also the case. If not this will be logged. Delete this comment if the log and the check if the log never appears.
+        // Check if /blob/ is not in the explodedPath. If so create a warning log.
+        if (str_contains($explodedPath, '/blob/') === false) {
+            $this->pluginLogger->warning('In this function we expect that a logo with host https://github.com always contains /blob/{branch} in the URL. We need to think about whether we want to support this URL or whether we want to include it in the documentation (if it is not already the case)', ['open-catalogi/open-catalogi-bundle']);
+        }
+
+        // Check if /blob/ is in the explodedPath.
+        if (str_contains($explodedPath, '/blob/') === true) {
+            // Explode the path with /.
+            // The first three items in the array is always:
+            // An empty string = 0, blob = 1 and the branch = 2. This has to be removed from the path.
+            $explodedPath = explode('/', $explodedPath);
+
+            // Loop till the total amount of the explodedPath array.
+            $path = null;
+            for ($i = 0; $i < count($explodedPath); $i++) {
+                // If i is 0, 1, 2 then nothing is done.
+                // The /blob/{branch} will not be added to the path.
+                if ($i === 0
+                    || $i === 1
+                    || $i === 2
+                ) {
```

Comment:
if (in_array($i, [0,1,2])) might look better and results in a 1 line if statement.
Or: 

if ($i <= 2) 
   continue;